kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/bluetooth/le/beacon/BluetoothLeBeacon.java

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Eurotech and/or its affiliates
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.bluetooth.le.beacon;
+
+/**
+ * BluetoothLeBeacon is a representation of a generic Beacon advertise packet.
+ * It offers methods to serialize the Beacon object.
+ *
+ */
+public abstract class BluetoothLeBeacon {
```

Comment:
We have fields that are common to all beacons, so I prefer to keep them here using an abstract class.